src/System.Management.Automation/engine/Modules/ModuleUtils.cs

```diff
@@ -399,11 +406,48 @@ internal static IEnumerable<CommandInfo> GetMatchingCommands(string pattern, Exe
                     }
 
                     string moduleShortName = System.IO.Path.GetFileNameWithoutExtension(modulePath);
-                    var exportedCommands = AnalysisCache.GetExportedCommands(modulePath, false, context);
+
+                    // If the module is on the System32 path where CompatiblePSEditions are checked,
+                    // we are forced to use Get-Module to discover the compatibility of the module
+                    string psCompatibleEditionsCheckedPath = ModuleIntrinsics.GetWindowsPowerShellPSHomeModulePath();
+                    if (modulePath.StartsWith(psCompatibleEditionsCheckedPath, StringComparison.OrdinalIgnoreCase))
+                    {
+                        // Do our best not to call Get-Module by using a cache
+                        if (s_incompatibleEditionSystem32Modules.ContainsKey(modulePath))
+                        {
+                            // We have already identified the module as incompatible
+                            continue;
+                        }
+                        else
+                        {
+                            using (PowerShell pwsh = PowerShell.Create())
+                            {
+                                PSModuleInfo module = pwsh.AddCommand("Get-Module")
```

Comment:
I quite dislike this -- not sure if there's a better way to do it?

Reply:
I see we have `private IEnumerable<PSModuleInfo> GetModuleForRootedPaths(string[] modulePaths, bool all, bool refresh)`